### ISSUE TITLE
js function cause the ui to redraw after level 2 depth in nested module

### ISSUE THREAD

**Issue by happyshows** [daattali/shinyjs#111]
Hi,

I was expecting to disable the clicked button in calling (parent) module after it got clicked exactly once. However, somehow it's triggering a redraw in the calling module which invalid the original purpose. I don't know if I have done anything wrong here. Thanks for the help.

minimal example
```
library(shiny)
library(shinyjs)

moduleTestUI <- function(id, label='ui1'){
  ns <- NS(id)
  shinyUI(fluidPage(
    uiOutput(ns('ui_test'))
  ))
}

moduleTest <- function(input, output, session,options) {
  ns <- session$ns
  
  ls_options <- lapply(options,function(opt){
    bsButton(ns(paste0('bb_',opt)),toupper(opt))
  })
  btn_ui <- do.call(tagList, ls_options)
  
  noshow <- lapply(options,function(opt){
    observe({
      req(input[[btn_id <- paste0('bb_',opt)]] ==1)
      isolate({
        disable(id = btn_id)
        insertUI(
          selector = "#placeHolder",
          where = "beforeEnd",
          ui = moduleTestUI(ns('ro'))
        )
        callModule(moduleTest,'ro',options = letters[sample(1:26,2)])
      })
    })
  })
  
  output$ui_test <- renderUI({
    
    isolate({
      btn_ui
    })
  })
}

ui <- tagList(useShinyjs(),
              fluidPage(
                uiOutput('page'),
                tags$div(id = 'placeHolder')
              ))

server <- function(input, output,session) {
  output$page <- renderUI({
    moduleTestUI('ui1')
  })
  callModule(moduleTest,'ui1',options = letters[sample(1:26,2)])
}

shinyApp(ui = ui, server = server)
```

**Comment by daattali** [comment]
Hmm when modules were released I knew that they it would be terrible for shinyjs..... I'm surprised it took this long for someone to find an issue with it :) 

This code doesn't run for me out of the box - can I safely replace `bsButton` with `actionButton`?

I don't quite understand what this code is trying to do, can you describe what the app does, what is the expected behaviour, and what is the current behaviour?

**Comment by daattali** [comment]
I was step-debuggin through the app and I noticed that the disable function DOES work. The second (and third and so on) set of buttons DO get disabled. But they also get overwritten (get generated again) , and when they get regenerated, they are initialized as normal (normal = not disabled).

You can see that's the case by adding a `Sys.sleep(1)` right after the `disable()` command, and you'll see that the button does get disabled. And to prove to you that the reason it gets "undisabled" is because it gets re-generated, try replacing the `disable()` function with `html(id = btn_id, html = "test")` - you'll see that after pressing the first row of buttons, the new text remains, but in any subsequent row the text changes back because the button is regenerated.

Phew! So shinyjs still has 0 bugs with modules :D

If you want to find a workaround.... I don't quite understand what the app is doing, but that's a shiny issue and not a shinyjs issue so I'm closing this issue. I hope you figure it out

**Comment by happyshows** [comment]
I was trying to create a decision tree model using shiny modules. So basically for each module instance, there are different decision variables (represented by a button with unique value for that context) and one of the variable will be clicked (only once) to call a nested module for the next decision module instance. 

x,y & x'y' are buttons inside module instances.
A - x - B - x' - C
A - y - D - y' - E

I need to call disable as I don't want user to click button twice.

when I was debugging earlier, I found out button disabled but the somehow the button itself got reverted to original status. I'll submit an issue in shiny github and see how they respond...
